qa/os/src/test/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -263,6 +263,64 @@ public void test60AutoCreateKeystore() throws Exception {
         });
     }
 
+    public void test61PasswordProtectedKeystore() throws Exception {
+        assumeThat(installation, is(notNullValue()));
+        // cleanup from previous test
+        // rm(installation.config("elasticsearch.keystore"));
+
+        String password = "keystorepass";
+        final Shell sh = new Shell();
+        Platforms.onLinux(() -> {
+            String systemJavaHome = sh.run("echo $SYSTEM_JAVA_HOME").stdout.trim();
+            sh.getEnv().put("JAVA_HOME", systemJavaHome);
+        });
+        Platforms.onWindows(() -> {
+            final String systemJavaHome = sh.run("$Env:SYSTEM_JAVA_HOME").stdout.trim();
+            sh.getEnv().put("JAVA_HOME", systemJavaHome);
+        });
+
+
+        final Installation.Executables bin = installation.executables();
+
+        // set the password by passing it to stdin twice
+        Platforms.onLinux(() ->
+            sh.run("echo $\'" + password + "\n" + password + "\n\' | sudo -u " + ARCHIVE_OWNER + " "
+                + bin.elasticsearchKeystore + " passwd")
+        );
+        Platforms.onWindows(() -> {
+            sh.run("echo \"" + password + "`r`n" + password + "`r`n\" | " + bin.elasticsearchKeystore + " passwd");
+        });
+
+        Archives.runElasticsearch(installation, sh, password);
+        ServerUtils.runElasticsearchTests();
+        Archives.stopElasticsearch(installation);
+    }
+
+    public void test62PasswordProtectedKeystoreIncorrectPassword() throws Exception {
+        assumeThat(installation, is(notNullValue()));
+
+        final Shell sh = new Shell();
+        Platforms.onLinux(() -> {
+            String systemJavaHome = sh.run("echo $SYSTEM_JAVA_HOME").stdout.trim();
+            sh.getEnv().put("JAVA_HOME", systemJavaHome);
+        });
+        Platforms.onWindows(() -> {
+            final String systemJavaHome = sh.run("$Env:SYSTEM_JAVA_HOME").stdout.trim();
+            sh.getEnv().put("JAVA_HOME", systemJavaHome);
+        });
+
+        RuntimeException expected = null;
+        try {
+            Archives.runElasticsearch(installation, sh, "wrong");
+        } catch (RuntimeException e) {
```

Comment:
We should create an assertStartFailure method or something like that instead of relying on exceptions for control flow.